src/transform/classify.rs

```diff
@@ -0,0 +1,69 @@
+/// Transforms for classifying (tagging and categorizing) events.
+///
+/// Based on code in aw_research: https://github.com/ActivityWatch/aw-research/blob/master/aw_research/classify.py
+
+use regex::Regex;
+use serde_json;
+
+use std::collections::HashSet;
+use crate::models::Event;
+
+pub fn classify(mut events: Vec<Event>, classes: Vec<(String, Regex)>) -> Vec<Event> {
+    let mut classified_events = Vec::new();
+    for event in events.drain(..) {
+        classified_events.push(classify_one(event, &classes));
+    }
+    return classified_events;
+}
+
+fn classify_one(mut event: Event, classes: &Vec<(String, Regex)>) -> Event {
+    let mut tags: HashSet<String> = HashSet::new();
+    for (cls, re) in classes {
+        for val in event.data.values() {
```

Comment:
I'm thinking out loud now:

Maybe there's an issue with going through all values by default? Maybe we should supply which fields in the event we want to match data with? Or maybe it's better to just filter the fields before calling classify?

Probably the latter now that I think of it...

Reply:
In the aw-research classifier I actually only match on the fields "title", "app", and "url". I couldn't decide on what was the right approach here so I settled for the thing that made the least assumptions.